Replace QuaZip (and other archiving operations) with libarchive (#3959)
PrismLauncher/PrismLauncher: CMakeLists.txt

    set(QT_LIBEXECS_DIR ${QT${QT_VERSION_MAJOR}_INSTALL_PREFIX}/${QT${QT_VERSION_MAJOR}_INSTALL_LIBEXECS})
endif()

find_package(PkgConfig)
# Find libarchive
## NOTE(@getchoo): Never use pkg-config with MSVC since the vcpkg port makes our install bundle fail to find the dll
if(PkgConfig_FOUND AND NOT MSVC)
    pkg_check_modules(libarchive REQUIRED IMPORTED_TARGET libarchive)
    set(LIBARCHIVE_LIBRARIES PkgConfig::libarchive)
else()
    find_path(LIBARCHIVE_INCLUDE_DIR archive.h REQUIRED)
    find_library(LIBARCHIVE_LIBRARY_RELEASE archive REQUIRED)

>     find_library(LIBARCHIE_LIBRARY_DEBUG archived)
>     set(LIBARCHIVE_LIBRARIES optimized ${LIBARCHIVE_LIBRARY_RELEASE} debug ${LIBARCHIVE_LIBRARY_DEBUG})

Reviewer: ```suggestion
    set(LIBARCHIVE_LIBRARIES optimized ${LIBARCHIVE_LIBRARY_RELEASE})
```
Seems libarchive doesn't generate these

    set(QT_LIBEXECS_DIR ${QT${QT_VERSION_MAJOR}_INSTALL_PREFIX}/${QT${QT_VERSION_MAJOR}_INSTALL_LIBEXECS})
endif()

> find_package(PkgConfig REQUIRED)

Reviewer: ```suggestion
find_package(PkgConfig)
```
This technically isn't required since we have a fallback

> # Find libqrencode
> ## NOTE(@getchoo): Never use pkg-config with MSVC since the vcpkg port makes our install bundle fail to find the dll
> if(PkgConfig_FOUND AND NOT MSVC)
>     pkg_check_modules(libarchive REQUIRED IMPORTED_TARGET libarchive)
>     set(LIBARCHIVE_LIBRARY PkgConfig::libarchive)

Reviewer: ```suggestion
# Find libarchive
## NOTE(@getchoo): Never use pkg-config with MSVC since the vcpkg port makes our install bundle fail to find the dll
if(PkgConfig_FOUND AND NOT MSVC)
    pkg_check_modules(libarchive REQUIRED IMPORTED_TARGET libarchive)
    set(LIBARCHIVE_LIBRARIES PkgConfig::libarchive)
```

else()

>     find_library(LibArchive REQUIRED)
>     set(LIBARCHIVE_LIBRARY LibArchive::LibArchive)

Reviewer: ```suggestion
    find_path(LIBARCHIVE_INCLUDE_DIR archive.h REQUIRED)
    find_library(LIBARCHIVE_LIBRARY_RELEASE archive REQUIRED)
    find_library(LIBARCHIE_LIBRARY_DEBUG archived)
    set(LIBARCHIVE_LIBRARIES optimized ${LIBARCHIVE_LIBRARY_RELEASE} debug ${LIBARCHIVE_LIBRARY_DEBUG})
```

endif()

if(NOT Launcher_FORCE_BUNDLED_LIBS)
    # Find toml++
    find_package(tomlplusplus 3.2.0 QUIET)

>     # Fallback to pkg-config if CMake files aren't found

Reviewer: ```suggestion
    # Fallback to pkg-config (if available) if CMake files aren't found
```

    if(NOT tomlplusplus_FOUND)

>         pkg_check_modules(tomlplusplus IMPORTED_TARGET tomlplusplus>=3.2.0)

Reviewer: ```suggestion
        if(PkgConfig_FOUND)
            pkg_check_modules(tomlplusplus IMPORTED_TARGET tomlplusplus>=3.2.0)
        endif()
```
Ditto

endif()

if(NOT Launcher_FORCE_BUNDLED_LIBS)
    find_package(PkgConfig REQUIRED)

>     pkg_check_modules(libarchive IMPORTED_TARGET libarchive)

Reviewer: This should also be taken out of the current conditional since no bundled version of this exists
```suggestion
    find_package(PkgConfig REQUIRED)
    pkg_check_modules(libarchive REQUIRED IMPORTED_TARGET libarchive)
```

endif()

if(NOT Launcher_FORCE_BUNDLED_LIBS)
    find_package(PkgConfig REQUIRED)
    pkg_check_modules(libarchive IMPORTED_TARGET libarchive)
    # Find toml++
    find_package(tomlplusplus 3.2.0 QUIET)

>     # Fallback to pkg-config (if available) if CMake files aren't found

Reviewer: ```suggestion
    # Fallback to pkg-config if CMake files aren't found
```

    # Find qrcodegencpp-cmake
    find_package(qrcodegencpp QUIET)

    find_package(LibArchive 3.7.8 QUIET)
    # Fallback to pkg-config (if available) if CMake files aren't found
    if(NOT LibArchive_FOUND)
        find_package(PkgConfig)
        if(PkgConfig_FOUND)
            pkg_check_modules(LibArchive IMPORTED_TARGET LibArchive>=3.7.8)
        endif()

>     endif()

Reviewer: ```suggestion
    find_package(PkgConfig REQUIRED)
    pkg_check_modules(libarchive REQUIRED IMPORTED_TARGET libarchive)
```
I don't know of any distribution of libarchive that has CMake support...
